File: __aacontest/today/3.py
from collections import defaultdict

class Solution:
    def mostFrequentIDs(self, nums, freq):
        counts = defaultdict(int)
        max_count = 0
        ans = []

        for i in range(len(nums)):
            counts[nums[i]] += freq[i]
            max_count = max(max_count, counts[nums[i]])
            ans.append(max_count if max_count > 0 else 0)

        return ans

# Test Cases
solution = Solution()
print(solution.mostFrequentIDs([2,3,2,1], [3,2,-3,1]))  # Output: [3, 3, 2, 2]
print(solution.mostFrequentIDs([5,5,3], [2,-2,1]))      # Output: [2, 0, 1]

File: __aacontest/today/3.rb
def min_operations(k)
  operations = 0
  nums = [1]
  
  while nums.reduce(:+) < k
    max_num = nums.max
    if nums.sum { |val| val * 2 } >= k
      nums[nums.index(max_num)] *= 2
      break
    else
      nums.map! { |num| num + 1 }
    end
    operations += 1
  end
  
  operations
end

# Test cases
puts min_operations(11) # Output: 5
puts min_operations(1)  # Output: 0
puts min_operations(2)  # Output: 1
